tests/functional/olp-cpp-sdk-dataservice-read/VolatileLayerClientTest.cpp

```diff
@@ -0,0 +1,229 @@
+#include <olp/core/logging/Log.h>
+#include <olp/core/porting/make_unique.h>
+
+#include <chrono>
+#include <iostream>
+#include <string>
+#include <testutils/CustomParameters.hpp>
+
+#include "olp/dataservice/read/PartitionsRequest.h"
+#include "olp/dataservice/read/VolatileLayerClient.h"
+
+using namespace olp::dataservice::read;
+using namespace testing;
+
+namespace {
+
+using namespace olp::dataservice::read;
+const auto kAppIdEnvName = "dataservice_read_test_appid";
+const auto kAppSecretEnvName = "dataservice_read_test_secret";
+const auto kCatalogEnvName = "dataservice_read_test_catalog";
+const auto kLayerEnvName = "dataservice_read_volatile_layer";
+const auto kTimeout = std::chrono::seconds(5);
+
+class VolatileLayerClientTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    auto network = olp::client::OlpClientSettingsFactory::
+        CreateDefaultNetworkRequestHandler();
+
+    const auto key_id = CustomParameters::getArgument(kAppIdEnvName);
+    const auto secret = CustomParameters::getArgument(kAppSecretEnvName);
+
+    olp::authentication::Settings authentication_settings({key_id, secret});
+    authentication_settings.network_request_handler = network;
+
+    olp::authentication::TokenProviderDefault provider(authentication_settings);
+    olp::client::AuthenticationSettings auth_client_settings;
+    auth_client_settings.provider = provider;
+    settings_ = olp::client::OlpClientSettings();
+    settings_.network_request_handler = network;
+    settings_.authentication_settings = auth_client_settings;
+    olp::cache::CacheSettings cache_settings;
+    settings_.cache = olp::client::OlpClientSettingsFactory::CreateDefaultCache(
+        cache_settings);
+  }
+
+  void TearDown() override {
+    auto network = std::move(settings_.network_request_handler);
+    // when test ends we must be sure that network pointer is not captured
+    // anywhere
+    EXPECT_EQ(network.use_count(), 1);
+  }
+
+  std::string GetTestCatalog() {
+    return CustomParameters::getArgument(kCatalogEnvName);
+  }
+
+  std::string GetTestLayer() {
+    return CustomParameters::getArgument(kLayerEnvName);
+  }
+
+ protected:
+  olp::client::OlpClientSettings settings_;
+};
+
+TEST_F(VolatileLayerClientTest, GetPartitions) {
+  olp::client::HRN hrn(GetTestCatalog());
+  {
+    SCOPED_TRACE("Get Partitions Test");
+
+    VolatileLayerClient client(hrn, GetTestLayer(), settings_);
+
+    olp::client::Condition condition;
+    olp::dataservice::read::VolatileLayerClient::PartitionsResponse
+        partitions_response;
+
+    auto callback =
+        [&](olp::dataservice::read::VolatileLayerClient::PartitionsResponse
+                response) {
+          partitions_response = std::move(response);
+          condition.Notify();
+        };
+    client.GetPartitions(olp::dataservice::read::PartitionsRequest(),
+                         std::move(callback));
+    ASSERT_TRUE(condition.Wait(kTimeout));
```

Comment:
Consider to use std::promise instead of custom condition class:
```
    std::promise<VolatileLayerClient::PartitionsResponse> promise;
    auto future = promise.get_future();
    auto token = client.GetPartitions(
        request, [&promise](VersionedLayerClient::PartitionsResponse response) {
          promise.set_value(response);
        });
    ASSERT_NE(future.wait_for(kWaitTimeout), std::future_status::timeout);
    auto partitions_response = future.get();
```